src/Foundatio.Storage.SshNet/Storage/SshNetFileStorage.cs

```diff
@@ -13,384 +14,432 @@
-                await Task.Factory.FromAsync(_client.BeginUploadFile(stream, path, null, null), _client.EndUploadFile).AnyContext();
-            }
+    public async Task<bool> CopyFileAsync(string path, string targetPath, CancellationToken cancellationToken = default) {
+        if (String.IsNullOrEmpty(path))
+            throw new ArgumentNullException(nameof(path));
+        if (String.IsNullOrEmpty(targetPath))
+            throw new ArgumentNullException(nameof(targetPath));
 
-            return true;
-        }
+        using var stream = await GetFileStreamAsync(path, cancellationToken).AnyContext();
+        if (stream == null)
+            return false;
 
-        public Task<bool> RenameFileAsync(string path, string newPath, CancellationToken cancellationToken = default) {
-            if (String.IsNullOrEmpty(path))
-                throw new ArgumentNullException(nameof(path));
-            if (String.IsNullOrEmpty(newPath))
-                throw new ArgumentNullException(nameof(newPath));
+        return await SaveFileAsync(targetPath, stream, cancellationToken).AnyContext();
+    }
 
-            newPath = NormalizePath(newPath);
-            EnsureClientConnected();
+    public Task<bool> DeleteFileAsync(string path, CancellationToken cancellationToken = default) {
+        if (String.IsNullOrEmpty(path))
+            throw new ArgumentNullException(nameof(path));
 
-            try {
-                _client.RenameFile(NormalizePath(path), newPath, true);
-            } catch (SftpPathNotFoundException) {
-                CreateDirectory(newPath);
-                _client.RenameFile(NormalizePath(path), newPath, true);
-            }
+        EnsureClientConnected();
 
-            return Task.FromResult(true);
+        string normalizedPath = NormalizePath(path);
+        _logger.LogTrace("Deleting {Path}", normalizedPath);
+        
+        try {
+            _client.DeleteFile(normalizedPath);
+        } catch (SftpPathNotFoundException ex) {
+            _logger.LogDebug(ex, "{Path} does not exist", normalizedPath);
         }
 
-        public async Task<bool> CopyFileAsync(string path, string targetPath, CancellationToken cancellationToken = default) {
-            if (String.IsNullOrEmpty(path))
-                throw new ArgumentNullException(nameof(path));
-            if (String.IsNullOrEmpty(targetPath))
-                throw new ArgumentNullException(nameof(targetPath));
-
-            using var stream = await GetFileStreamAsync(path, cancellationToken).AnyContext();
-            if (stream == null)
-                return false;
+        return Task.FromResult(true);
+    }
 
-            return await SaveFileAsync(targetPath, stream, cancellationToken).AnyContext();
-        }
+    public async Task<int> DeleteFilesAsync(string searchPattern = null, CancellationToken cancellationToken = default) {
+        using var _ = _logger.BeginScope(s => s.Property("SearchPattern", searchPattern));
```

Comment:
Why are you starting a logging scope here? You only do logging scopes if you want every single log message inside of the scope to all have the same properties. I don't see any benefit to having prefix and pattern on every single message. You can log out the prefix and pattern at the start of the operation.

Reply:
Because I don't want to log it out in every log message but I want to see what the pattern was for the delete.

Reply:
Done